hadoop-common-project/hadoop-common/src/main/proto/RpcHeader.proto

```diff
@@ -91,6 +91,7 @@ message RpcRequestHeaderProto { // the header for the RpcRequest
   optional RPCTraceInfoProto traceInfo = 6; // tracing info
   optional RPCCallerContextProto callerContext = 7; // call context
   optional int64 stateId = 8; // The last seen Global State ID
+  optional bytes routerFederatedState = 9; // Alignment context info for use with routers.
```

Comment:
I think you should add a comment here that says that the bytes should not be interpreted by the client and should only be come from RpcResponseHeaderProto.routerFederatedState.

Reply:
Added comments.